### ISSUE TITLE
Automatic OIDC (SSO) Login is not working (with Keycloak)

### ISSUE THREAD

**Issue by lsauer** [DependencyTrack/dependency-track#1855]
The enhancement may already be reported! Please search for the enhancement before creating one.

### Current Behavior:
When logged in by OIDC (SSO) and then going on the DT 4.5 page an automatic login attempted however it is apparently sending something to the local login endpoint.

### Proposed Behavior:
DT should recognize that the current session is an OIDC and automatically attempt a OIDC login instead of a local login.

**Comment by mulder999** [comment]
I agree with the suggestion but maybe your title should be updated to something like "Improvement: Allow to automatically redirect to SSO login when OIDC is configured"

Replacing local login by the remote login would be an interesting feature when OIDC is configured. That would avoid an extra click for the user.

**Comment by syalioune** [comment]
> DT should recognize that the current session is an OIDC and automatically attempt a OIDC login instead of a local login.

I believe this is a misconception about how DT works. DT would indeed recognize the SSO session and the subsequent call to `/api/v1/login` is not to perform a redundant local login but to generate and internal JWT token that contains the user privileges. External OIDC provider is only used for authentifcation and DT still handle authorization.

> Replacing local login by the remote login would be an interesting feature when OIDC is configured. That would avoid an extra click for the user.

There already [frontend issue #429](https://github.com/DependencyTrack/frontend/issues/429) to track that enhancement

**Comment by Nikemare** [comment]
This issue relates to [frontend issue #530](https://github.com/DependencyTrack/frontend/issues/530)

**Comment by slaib** [comment]
Dear Maintainer,
I tried to get DependencyTrack version 4.13.2 and Keycloak 26.2.4 to work together, i.e., the SSO feature. Both systems, DependencyTrack and Keycloak, run without any problems.

Unfortunately, it always fails on the DependencyTrack side, and the SSO feature is not offered on the login page. DependencyTrack works in combination with a very old Keycloak version, e.g., 16.1.1.

What's the situation here? Will you further develop or adapt this feature in DependencyTrack, including your documentation, so that it works again with the current Keycloak version?

Thank you in advance for your feedback and effort.

**Comment by slaib** [comment]
@mulder999 Thank you very much for your fast response.
I was also seeing that Keycloak changed it that `/auth` is no longer part of the base URL.
My setup: I use as Docker container setup Keycloak and DependencyTrack + Caddy as reverse proxy just with the standard config without special configuration.

In Keycloak I set the following two "important" env variables:
`KC_HOSTNAME=auth.localhost`        # just my setup
`KC_HTTP_RELATIVE_PATH=/auth`       # this makes it possible to have `/auth` in the base URL

I tested and checked it again also with your information --> **with** `/auth` and also **without** **/auth** in the base URL.
But I get every time the error output in the log (see also the attached pictures): `ERROR [OidcConfigurationResolver] Failed to fetch OIDC configuration from issuer ...`
I will attach a few screenshots here to this comment.

Do you have any other ideas on what I need to do to make DependencyTrack and Keycloak work together? Do you personally have a working setup? 

Thank you in advance for your feedback and effort.

![Image](https://github.com/user-attachments/assets/01e37195-812d-432e-a8ce-fba2db278cd4)

![Image](https://github.com/user-attachments/assets/ec3a1269-ef2a-4f8a-a9bc-a6ec28d2b9df)

![Image](https://github.com/user-attachments/assets/d8be53fc-ecc1-495d-9fe1-681d68d2d0da)

![Image](https://github.com/user-attachments/assets/d576c2e2-ddca-4dd0-b4c2-71641ac638f9)

**Comment by mulder999** [comment]
@slaib I recommend bringing this up in the discussion channel, as your issue doesn't appear to be related to this topic nor a general Dependency-Track and OIDC integration problem. The setup you're using is known to work correctly—we provide Dependency-Track as a managed service alongside Keycloak for our customers without issue.

In your case, it would be worth investigating whether the untrusted or invalid certificate is contributing to the issue, as this can interfere with OIDC discovery and authentication. Addressing the certificate trust problem might help resolve the behavior you're seeing.
